presto-main/src/main/java/com/facebook/presto/sql/planner/DependencyExtractor.java

```diff
@@ -43,6 +44,14 @@ private DependencyExtractor() {}
         return uniqueSymbols.build();
     }
 
+    public static Set<Symbol> extractUniqueNonRecursive(PlanNode node)
```

Comment:
I think the calling site needs to inspect node non-recursively to identify situation when

- node _is_ correlated
- sources are _not_ correlated

But maybe I just don't understand.